syntax = "proto3";

package skaiproto.feetpos;

// extra metadata for other messages to use 
message FeetPosition {
    // xyz float value in meters
    float x = 1;
    float y = 2;
    float z = 3;
    
    // metadata only used by other messages
    uint64 timestamp = 4;
    uint64 skaimot_id = 5;
}

message FeetPosProtoMsg {

    message CameraFrame {
        
        // camera mac address turned into serial num
        uint64 camera_id = 1;

        // 1e9 * epoch timestamp
        uint64 timestamp = 2;

        // people in frame
        message Person {

            // skaimot id num
            uint64 id = 1;

            // feet position xyz float meters
            FeetPosition feet_position = 2;

            // confidence based on within edge of frame or not
            float confidence = 3;
        }
        repeated Person people_in_frame = 3;

    }
    repeated CameraFrame camera_frames = 1;

}